### DISCUSSION TITLE
The easiest way to log api response from serverless function?

### DISCUSSION THREAD

**Discussion by greggf** [vercel/community#75] [Help] [answered]
I want to keep a log of api responses to keep an eye on errors and audit lead generation data, however serverless functions do not allow me to write to files.  

What is the easiest way to log and access this data?  I've been told headless CMS, but that seems like overkill.  There has to be an easier solution.

**Comment by leerob** [comment, marked as answer]
You can use any [logging integration](https://vercel.com/integrations#logging) to send logs from Vercel to any provider of your choice for storage and alerting. [Learn more about logs on Vercel](https://vercel.com/docs/concepts/deployments/logs).
